cli/azd/internal/vsrpc/aspire_service.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package vsrpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/internal/appdetect"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/dotnet"
+	"github.com/golobby/container/v3"
+)
+
+// aspireService is the RPC server for the '/aspireService/v1.0' endpoint.
+type aspireService struct {
+	server *Server
+}
+
+func newAspireService(server *Server) *aspireService {
+	return &aspireService{
+		server: server,
+	}
+}
+
+// GetAspireHostAsync is the server implementation of:
+// ValueTask<AspireHost> GetAspireHostAsync(Session session, string aspireEnv, CancellationToken cancellationToken).
+func (s *aspireService) GetAspireHostAsync(
+	ctx context.Context, sessionId Session, aspireEnv string, observer IObserver[ProgressMessage],
+) (*AspireHost, error) {
+	session, err := s.server.validateSession(ctx, sessionId)
+	if err != nil {
+		return nil, err
+	}
+
+	session.sessionMu.Lock()
+	defer session.sessionMu.Unlock()
+
+	var c struct {
+		azdContext *azdcontext.AzdContext `container:"type"`
+		dotnetCli  dotnet.DotNetCli       `container:"type"`
+	}
+
+	if err := session.container.Fill(&c); err != nil {
+		return nil, err
+	}
+
+	// If there is an azure.yaml, load it and return the services.
+	if _, err := os.Stat(c.azdContext.ProjectPath()); err == nil {
+		var cc struct {
+			projectConfig *project.ProjectConfig `container:"type"`
+		}
+
+		if err := container.Fill(&cc); err != nil {
+			return nil, err
+		}
+
+		if session.appHostPath == "" {
+			appHost, err := appHostForProject(ctx, cc.projectConfig, c.dotnetCli)
+			if err != nil {
+				return nil, err
+			}
+
+			session.appHostPath = appHost.Path()
+		}
+
+		hostInfo := &AspireHost{
+			Name: filepath.Base(filepath.Dir(session.appHostPath)),
+			Path: session.appHostPath,
+		}
+
+		manifest, err := session.readManifest(ctx, session.appHostPath, c.dotnetCli)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load app host manifest: %w", err)
+		}
+
+		hostInfo.Services = servicesFromManifest(manifest)
+
+		return hostInfo, nil
+
+	} else if errors.Is(err, os.ErrNotExist) {
+		hosts, err := appdetect.DetectAspireHosts(ctx, c.azdContext.ProjectDirectory(), c.dotnetCli)
+		if err != nil {
+			return nil, fmt.Errorf("failed to discover app host project under %s: %w", c.azdContext.ProjectPath(), err)
+		}
+
+		if len(hosts) == 0 {
+			return nil, fmt.Errorf("no app host projects found under %s", c.azdContext.ProjectPath())
+		}
+
+		if len(hosts) > 1 {
```

Comment:
I think this is out of scope for V1 but yes it is something we should think about at some point - there's an interesting question as to what it would mean to have two Aspire apps in the same `azd` project.  I'm wondering what, if any resources they would share.

The answer may differ between target compute platforms.  For example, I could imagine with AKS we try to share the cluster and use a separate namespace for each app to provide some level of isolation?

Reply:
Put https://github.com/Azure/azure-dev/issues/3289 in the backlog and assigned it to me so I remember to think about it more.